db/memtable.cc

```diff
@@ -416,24 +417,36 @@ InternalIterator* MemTable::NewIterator(const ReadOptions& read_options,
   return new (mem) MemTableIterator(*this, read_options, arena);
 }
 
+void MemTable::RebuildFragmentedTombstones(const ReadOptions& read_options) {
+  auto* unfragmented_iter = new MemTableIterator(
+      *this, read_options, nullptr /* arena */, true /* use_range_del_table */);
+
+  if (unfragmented_iter != nullptr) {
+    auto new_fragmented_tombstones_list =
+      std::make_shared<FragmentedRangeTombstoneList>(
```

Comment:
Do you only want to do this work when `fragmented_tombstones_list == nullptr`? I haven't fully thought through the implications but the current code appears to be doing the fragmenting work on every read.

Reply:
Whoops.  I've fixed it to only rebuild the fragmented tombstones when they are invalidated.